lib/spack/spack/build_systems/python.py

```diff
@@ -471,3 +471,10 @@ def remove_files_from_view(self, view, merge_map):
                 view.remove_file(src, dst)
             else:
                 os.remove(dst)
+
+    def setup_environment(self, spack_env, run_env):
+        """Set up the compile and runtime environments for a package.
+           Set PYTHONPATH if package is specified as external.
+	"""
+        if self.spec.external:
```

Comment:
Shouldn't we check that the package is external and not activated? I.e. install dir != `/activated`?